osquery/tables/system/windows/certificates.cpp

```diff
@@ -307,6 +311,230 @@ void parseSystemStoreString(LPCWSTR sysStoreW,
+          ? INTEGER(1)
+          : INTEGER(0);
+
+  r["not_valid_before"] =
+      INTEGER(filetimeToUnixtime(certContext->pCertInfo->NotBefore));
+
+  r["not_valid_after"] =
+      INTEGER(filetimeToUnixtime(certContext->pCertInfo->NotAfter));
+
+  r["signing_algorithm"] =
+      cryptOIDToString(certContext->pCertInfo->SignatureAlgorithm.pszObjId);
+
+  r["key_algorithm"] = cryptOIDToString(
+      certContext->pCertInfo->SubjectPublicKeyInfo.Algorithm.pszObjId);
+
+  r["key_usage"] = getKeyUsage(certContext->pCertInfo);
+
+  r["key_strength"] = INTEGER(
+      (certContext->pCertInfo->SubjectPublicKeyInfo.PublicKey.cbData) * 8);
+
+  certBuff.clear();
+  getCertCtxProp(certContext, CERT_KEY_IDENTIFIER_PROP_ID, certBuff);
+  std::string subjectKeyId;
+  boost::algorithm::hex(std::string(certBuff.begin(), certBuff.end()),
+                        back_inserter(subjectKeyId));
+  r["subject_key_id"] = subjectKeyId;
+
+  r["path"] =
+      storeLocation + "\\" + constructDisplayStoreName(storeId, storeName);
+  r["store_location"] = storeLocation;
+  r["store"] = storeName;
+
+  std::string serial;
+  boost::algorithm::hex(
+      std::string(certContext->pCertInfo->SerialNumber.pbData,
+                  certContext->pCertInfo->SerialNumber.pbData +
+                      certContext->pCertInfo->SerialNumber.cbData),
+      back_inserter(serial));
+  r["serial"] = serial;
+
+  std::string authKeyId;
+  if (certContext->pCertInfo->cExtension != 0) {
+    auto extension = CertFindExtension(szOID_AUTHORITY_KEY_IDENTIFIER2,
+                                       certContext->pCertInfo->cExtension,
+                                       certContext->pCertInfo->rgExtension);
+    if (extension != nullptr) {
+      unsigned long decodedBuffSize = 0;
+      CryptDecodeObjectEx(CERT_ENCODING,
+                          X509_AUTHORITY_KEY_ID2,
+                          extension->Value.pbData,
+                          extension->Value.cbData,
+                          CRYPT_DECODE_NOCOPY_FLAG,
+                          nullptr,
+                          nullptr,
+                          &decodedBuffSize);
+
+      certBuff.resize(decodedBuffSize, 0);
+      std::fill(certBuff.begin(), certBuff.end(), 0);
+      auto decodeRet = CryptDecodeObjectEx(CERT_ENCODING,
+                                           X509_AUTHORITY_KEY_ID2,
+                                           extension->Value.pbData,
+                                           extension->Value.cbData,
+                                           CRYPT_DECODE_NOCOPY_FLAG,
+                                           nullptr,
+                                           certBuff.data(),
+                                           &decodedBuffSize);
+      if (decodeRet != FALSE) {
+        auto authKeyIdBlob =
+            reinterpret_cast<CERT_AUTHORITY_KEY_ID2_INFO*>(certBuff.data());
+
+        boost::algorithm::hex(std::string(authKeyIdBlob->KeyId.pbData,
```

Comment:
Can this be factored out into an `inline` method, since it's used multiple times?

Reply:
I think so! FWIW, this was not code introduced in this PR, it was just code that was refactored out into a separate function. Do you think that kind of change would belong in this PR, or maybe a separate one?

Reply:
done 👍